internal/legacy/pinhal_baremetal.go

```
@@ -0,0 +1,33 @@
+//go:build baremetal
+
+package legacy
+
+import "machine"
+
+func configurePinOut(p PinOutput) {
+	configurePin(p, machine.PinOutput)
+}
+
+func configurePinInputPulldown(p PinInput) {
+	configurePin(p, pulldown) // some chips do not have pull down, in which case pulldown==machine.PinInput.
+}
+
+func configurePinInput(p PinInput) {
+	configurePin(p, machine.PinInput)
+}
+
+func configurePinInputPullup(p PinInput) {
+	configurePin(p, pullup) // some chips do not have pull up, in which case pullup==machine.PinInput.
+}
+
+func pinIsNoPin(a any) bool {
+	p, ok := a.(machine.Pin)
+	return ok && p == machine.NoPin
+}
+
+func configurePin(p any, mode machine.PinMode) {
+	machinePin, ok := p.(machine.Pin)
+	if ok {
```

What happens if "!ok" later on in the program? Should we return an error or at least print a message? Because pin configuration itself can fail if not "baremetal" I prefer to return an error. So the error can be handled at caller site, e.g. by retry.

---

Two things to keep in mind:
- If !ok then the user is using the driver with a pin they configured as per the new API agreement
- If a new driver is developed it will NOT use configurePin. this is legacy and will eventually be phased out over time.